packages/react-dom/src/test-utils/ReactTestUtilsAct.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import warningWithoutStack from 'shared/warningWithoutStack';
+
+const [
+  /* eslint-disable no-unused-vars */
+  _getInstanceFromNode,
+  _getNodeFromInstance,
+  _getFiberCurrentPropsFromNode,
+  _injectEventPluginsByName,
+  _eventNameDispatchConfigs,
+  _accumulateTwoPhaseDispatches,
+  _accumulateDirectDispatches,
+  _enqueueStateRestore,
+  _restoreStateIfNeeded,
+  _dispatchEvent,
+  _runEventsInBatch,
+  /* eslint-enable no-unused-vars */
+  setIsActingUpdatesInDev,
+] = ReactDOM.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.Events;
+
+// for .act's return value
+type Thenable = {
+  then(resolve: () => void, reject?: () => void): void,
+  // ^ todo - maybe this should return a promise...
+};
+
+// a stub element, lazily initialized, used by act() when flushing effects
+let actContainerElement = null;
+// a step counter when we descend/ascend from .act calls
+const actingScopeDepth = {current: 0};
+
+export default function act(callback: () => void | Promise<void>): Thenable {
```

Comment:
What about the test renderer?

Reply:
![image](https://user-images.githubusercontent.com/18808/52813917-c7fb3b80-3092-11e9-9365-d6b4b0806158.png)
working on it!